cmd/cosign/cli/initialize/init.go

```diff
@@ -35,7 +37,12 @@ func DoInitialize(ctx context.Context, root, mirror string, threshold int) error
 	}
 
 	// Initialize the remote repository.
-	remote, err := tuf.GcsRemoteStore(ctx, mirror, nil, nil)
+	var remote client.RemoteStore
+	if strings.HasPrefix(mirror, "http://") || strings.HasPrefix(mirror, "https://") {
```

Comment:
Idea: Perhaps you could use  `u, err := url.Parse(mirror)` to validate the mirror points to a valid URL, or simply check whether the mirror is `sigstore-tuf-root`. 

cc @asraa

Reply:
+1 preference to parse the URL to check if it's valid.

I would caution against hardcoding a check on whether it's `sigstore-tuf-root` though.

Reply:
Done